sdks/beam-transforms/src/main/java/org/gbif/pipelines/transforms/core/GrscicollTransform.java

```diff
@@ -96,21 +107,31 @@ public void tearDown() {
   }
 
   @Override
-  public Optional<GrscicollRecord> convert(ExtendedRecord source) {
+  public Optional<GrscicollRecord> convert(CoGbkResult source) {
     throw new IllegalArgumentException("Method is not implemented!");
   }
 
   @Override
   @ProcessElement
   public void processElement(ProcessContext c) {
-    processElement(c.element(), c.sideInput(metadataView)).ifPresent(c::output);
+    CoGbkResult v = c.element();
+
+    ExtendedRecord er = v.getOnly(erTag, null);
+    BasicRecord br = v.getOnly(brTag, null);
+
+    if (er == null || br == null) {
+      return;
+    }
```

Comment:
Should this check be in processElement? If so then it will work for beam and java versions